### DISCUSSION TITLE
I cannot boot my TV box H96-max-x3 with the Amlogic boot scripts for Armbian, but your debian system boots perfect

### DISCUSSION THREAD

**Discussion by rebe1956** [devmfc/amlogic-bootscripts-Armbian#3] [Q&A]
@devmfc:
I own a TV box of the type "H96-max-x3" with an S905X3 CPU for over a year and would like to port Armbian as the operating system to it.
In the more than 100 attempts I've made so far, I've stuck to this description --> "https://github.com/833M0L3/H96MaxX3".
I was unable to boot the system from the SD card or a USB stick.
I tested different image releases and different dtb files. Everything didn't work.

I was about to give up. Then I accidentally came across your project "Debian Linux image for Android TV boxes with Amlogic SOC's" (
https://github.com/devmfc/debian-on-amlogic ).
The quality of your project is amazing! Thank you for this perfect job!
I was able to immediately boot the Debian system from the SD card.
Everything I've tested works so far!
I don't understand why there are hundreds of posts in the Armbian forum about how to port Armbian to a TV box,
if it's that easy with your method.
Could you please do me a favor?
Since your project is a "private" project, I would like to port Armbian to my box.
I followed your description at "https://github.com/devmfc/amlogic-bootscripts-Armbian" exactly.
Unfortunately, the system does not boot from the SD card or a USB stick.
Now I'm wondering how can it be that the debian image boots smoothly on my box using your method, but the bookwrom image doesn't.
Thank you very much for your support.

**Comment by devmfc** [comment]
Thanks, you're welcome.

I don't understand what you mean; Bookworm is Debian? My images are Debian Bookworm and Ubuntu Noble at this moment? Doesn't my Bookworm image boot?

I don't know why Armbian does not boot on your box. I only know that Armbian does not officially support tv boxes and that the problem is most probably not in the Userspace part (Debian/Ubuntu) but in the kernel, DTB and/or bootloader.

**Reply by rebe1956** [reply]
Sorry. The description of my concerns was probably a bit confusing.
I'll try to do better in a second attempt.
As described, I was able to install your image "Debian Bookworm (v12) with Linux kernel 6.6.29" on my H96-max-x3-tv-box without any problems.
Since your image is a private project, I would like to install Armbian (for example "Armbian_jammy_save_2024.05") on my box.
Unfortunately I cannot boot the system from SD card. The boot method used in this image does not work on my box.
And now my question:
Since the "chain loaded u-boot" method doesn't work on my box, I would be interested to know whether the "vendor u-boot" method could also be used here, as with your image?
Many thanks for your response.

**Reply by devmfc** [reply]
AFAIK Armbian_jammy_save_2024.05 is NOT an Armbian image, it is an ophub image, based on Armbian. The Armbian team really does not like them to abuse their name.

I tested a real Armbian image yesterday (jammy minimal) with the scripts in this repo, with their x96-air dtb and it just works.

By the way, with Amlogic vendor bootloader, in my experience, you can better try to boot with a USB flash drive. I almost never use sd-cards, they are just an unreliable pain in the A...

**Reply by rebe1956** [reply]
Thank you for the tip about to boot with a USB flash drive. I will do the same in the future.
Could you please send me a link to the real Armbian image you used to test your box.
I would then test it on my H96-max-x3 box with the scripts in this repo.

Are the scripts in aml_autoscript and s905_autoscript identical?

Thank you very much in advance for your support.

**Reply by devmfc** [reply]
Sorry, was a Noble image, but that should not matter, if it boots it will most likely boot any Debian derivative.

https://github.com/armbian/community/releases/download/24.5.0-trunk.667/Armbian_community_24.5.0-trunk.667_Aml-s9xx-box_noble_current_6.6.31_minimal.img.xz

The scripts aml_autoscript and s905_autoscript are NOT identical?

But any way, I can not / do not have the time to support Armbian and/or Ophub images, your chances of getting support are better in their respective support forums. Sorry

**Reply by rebe1956** [reply]
It works!!
Thank you very much for your support. I couldn't have done it without your help.
Thank you also for the time you spent for me.

**Comment by LucasRay30** [comment]
Hello my friend!

I'm going to take advantage of this thread to ask a question.
I have a 905X3 and it works normally with the box=s905x3_generic in the Ubuntu-Resolute_6.18.10 image.
However, with Debian-Trixie_6.18.10, editing the boot.config with the same instruction, the system doesn't start.
If possible, could you check this?
Thank you in advance for the work you've done!

Best regards!
